### ISSUE TITLE
resolutionStrategy.componentSelection.all requires non intutive call to be used

### ISSUE THREAD

**Issue by JLLeitschuh** [gradle/gradle-script-kotlin#196]
The issue is detailed here:
https://github.com/gradle/gradle/issues/987

**TL;DR** This is a very non intuitive API call:

```kotlin
val schemaLookup = configurations.create("customizations-schema-lookup") {
    it.resolutionStrategy
        .componentSelection
        .all(object : Action<ComponentSelection> {
            @Mutate
            override fun execute(selection : ComponentSelection) {
                println("Executing")
                println(selection)
            }
        })
}
```

**Comment by bamboo** [comment]
Blocked by #8.

**Comment by JLLeitschuh** [comment]
I mean, this isn't quite superseded, isn't the goal for the JVM to call the `Action`?
So the underlying problem of needing to annotate the action method with a `@Mutate` annotation is a simple fix.

**Comment by bamboo** [comment]
The Kotlin compiler is selecting the wrong overload. It should be selecting the `Action` taking version but it's selecting the `Object` taking one which expects a rule source argument, that's why adding `@Mutate` fixed the problem.

**Comment by bamboo** [comment]
Proven to be fixed in 4eba4f44c97a768f16c50c1e3842a115a22c2489.

**Comment by peter-evans** [comment]
I'm experiencing this issue. Not sure if I'm doing something wrong or if this is still a bug.

Please see this branch for where I'm configuring a custom resolution strategy.
https://github.com/peter-evans/gradle-auto-dependency-updates/blob/exclusion-rules/build.gradle.kts#L62-L71

If I change the above code to the following, it fails with the same error described in this issue.
```
configurations.all {
    resolutionStrategy
        .componentSelection
        .all({ selection : ComponentSelection ->
            excludeSelectionByRegex(selection, "org\\.jetbrains.*", ".*", ".*-(eap|M).*")
        })
}
```

**Comment by JLLeitschuh** [comment]
@peter-evans can you open an new issue against gradle/gradle for this?

**Comment by StefMa** [comment]
Can't you move this issue? 🤔 😁
